File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const clothCnt = new Array(n).fill(1);
    for (let i = 0; i < n; i++) {
        if (lost.indexOf(i + 1) > -1) clothCnt[i]--;
        if (reserve.indexOf(i + 1) > -1) clothCnt[i]++;
    }
    const answer = clothCnt.forEach((v, idx) => {
        if (!v) {
            if (idx && clothCnt[idx - 1] > 1) {
                clothCnt[idx - 1]--;
                clothCnt[idx]++;
            } else if (idx !== n - 1 && clothCnt[idx + 1] > 1) {
                clothCnt[idx + 1]--;
                clothCnt[idx]++;
            }
        }
    });
    return clothCnt.reduce((acc, v) => acc + (v > 0 ? 1 : 0), 0);
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] counts = new int[n + 2];
        
        for (int i = 1; i <= n; i++) {
            counts[i] = 1;
        }
        
        for (int num : lost) {
            counts[num]--;
        }
        
        for (int num : reserve) {
            counts[num]++;
        }
        
        for (int i = 1; i <= n; i++) {
            if (counts[i] == 0) {
                if (counts[i - 1] > 1) {
                    counts[i - 1]--;
                    counts[i]++;
                } else if (counts[i + 1] > 1) {
                    counts[i + 1]--;
                    counts[i]++;
                }
            }
        }
        
        int answer = 0;
        for (int i = 1; i <= n; i++) {
            if (counts[i] > 0) answer++;   
        }
        return answer;
    }
}